packages/fincaps/src/batchQuery.js

```diff
@@ -0,0 +1,208 @@
+// @ts-check
+/** @typedef {'children' | 'data'} AgoricChainStoragePathKind */
+/** @template T @typedef {import('@endo/marshal').FromCapData<T>} FromCapData<T> */
+
+/**
+ * @param {[kind: AgoricChainStoragePathKind, item: string]} path
+ */
+export const pathToKey = path => path.join('.');
+
+/** @param {string} key */
+export const keyToPath = key => {
+  const [kind, ...rest] = key.split('.');
+  assert(kind === 'children' || kind === 'data');
+  /** @type {[kind: 'children' | 'data', item: string]} */
+  const out = [kind, rest.join('.')];
+};
+
+/** @param {string | unknown} d */
+const parseIfJSON = d => {
+  if (typeof d !== 'string') return d;
+  try {
+    return JSON.parse(d);
+  } catch {
+    return d;
+  }
+};
+
+/**
+ * @param {string} apiURL
+ * @param {object} io
+ * @param {typeof fetch} io.fetch
+ */
+export const makeLCD = (apiURL, { fetch }) => {
+  assert.typeof(apiURL, 'string');
+
+  /**
+   * @param {string} href
+   * @param {object} [options]
+   * @param {Record<string, string>} [options.headers]
+   */
+  const getJSON = (href, options = {}) => {
+    const opts = {
+      keepalive: true,
```

Comment:
What does LCD stand for?

Also, curious why you are choosing to use `keepalive`?  My MDN search is not providing helpful clues for this context

> The keepalive option can be used to allow the request to outlive the page. Fetch with the keepalive flag is a replacement for the [Navigator.sendBeacon()](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon) API.

Reply:
LCD = [Light client daemon](https://tutorials.cosmos.network/tutorials/1-tech-terms/#light-client-daemon-lcd)

I copied `keepAlive` from something else... IOU some git spelunking.